autonomy/analyse/service.py

```diff
@@ -168,19 +171,62 @@
         "service_registry_address",
         "on_chain_service_id",
         "share_tm_config_on_startup",
-        "tendermint_p2p_url",
+        "use_termination",
         "setup",
     ],
 }
 
 ABCI = "abci"
 LEDGER = "ledger"
 
-UNKNOWN_LEDGER_RE = re.compile(r".*\(\'(.*)\' was unexpected\)")
-ABCI_CONNECTION_VALIDATOR = Draft4Validator(schema=ABCI_CONNECTION_SCHEMA)
-LEDGER_CONNECTION_VALIDATOR = Draft4Validator(schema=LEDGER_CONNECTION_SCHEMA)
-ABCI_SKILL_VALIDATOR = Draft4Validator(schema=ABCI_SKILL_SCHEMA)
-ABCI_SKILL_MODEL_PARAMS_VALIDATOR = Draft4Validator(
+REQUIRED_PROPERTY_RE = re.compile(r"'(.*)' is a required property")
+PROPERTY_NOT_ALLOWED_RE = re.compile(
+    r"Additional properties are not allowed \((('(.*)'(,)? )+)(was|were) unexpected\)"
+)
+
+
+class CustomSchemaValidator(Draft4Validator):
+    """Custom schema validator to report all missing fields at once."""
+
+    def validate(self, *args: Any, **kwargs: Any) -> None:
+        """Validate and raise all errors at once."""
+        missing = []
+        extra = []
+        for error in self.iter_errors(*args, **kwargs):
+            message = cast(ValidationError, error).message
+            missing_property_check = REQUIRED_PROPERTY_RE.match(message)
+            if missing_property_check is not None:
+                (missing_property,) = cast(re.Match, missing_property_check).groups()
+                missing.append(missing_property)
+                continue
+
+            property_not_allowed_check = PROPERTY_NOT_ALLOWED_RE.match(message)
+            if property_not_allowed_check is not None:
+                extra += re.findall(
+                    "[a-zA-Z0-9]+", property_not_allowed_check.groups()[0]
+                )
+                continue
+
+            if "does not have enough properties" in message:
+                raise error
+
+        if len(extra) > 0:
+            raise ValidationError(
+                message="Additional properties found "
+                + ", ".join(map(lambda x: f"'{x}'", extra))
+            )
+
+        if len(missing) > 0:
+            raise ValidationError(
```

Comment:
two dedicated raises here (actually 3 in the function body)
looks like does not meet requirement show all the errors at once, but show all the same group errors at once